src/configs/FirebaseConfig.js

```diff
@@ -160,6 +183,42 @@ const FirebaseConfig = () => {
     if (postId) checkIsAuthorFollowingMe(postId);
   };
 
+  const handleDeepLinkPost = async (deepLinkUrl) => {
+    const postId = deepLinkUrl?.split('helio.social/post/')[1];
+    if (DEEPLINK_POST_REGEX.test(deepLinkUrl) && postId) checkIsAuthorFollowingMe(postId);
+  };
+
+  const handleDeepLinkUser = async (deepLinkUrl) => {
+    const username = deepLinkUrl?.split('helio.social/')[1];
+    if (username?.toLocaleLowerCase() === profile?.username?.toLocaleLowerCase()) {
+      return navigation.navigate('Profile');
+    }
+
+    if (DEEPLINK_USER_REGEX.test(deepLinkUrl) && username) {
+      try {
+        const replacedUsername = username?.replace('+', '');
+        const data = {
+          username: replacedUsername,
+          user_id: signedProfileId
+        };
+        handleMovePage(USER, data);
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  };
+
+  const handleDeepLinkCommunity = async (deepLinkUrl) => {
+    const communityName = deepLinkUrl?.split('helio.social/c/')[1];
+    if (DEEPLINK_COMMUNITY_REGEX.test(deepLinkUrl) && communityName) {
+      try {
+        navigation.navigate('TopicPageScreen', {id: communityName});
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  };
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `handleDeepLinkPost`, `handleDeepLinkUser`, and `handleDeepLinkCommunity` functions, you're using the `split` method to extract the post ID, username, or community name from the URL. This approach might fail if the URL contains additional paths or parameters. Consider using a more robust method to parse the URL, such as the URL API.

Also, in the `handleDeepLinkUser` function, you're replacing '+' characters in the username with an empty string. This could potentially alter the actual username if it contains '+' characters. If you're trying to decode URL-encoded characters, consider using `decodeURIComponent` instead.

```diff
  const handleDeepLinkPost = async (deepLinkUrl) => {
-   const postId = deepLinkUrl?.split('helio.social/post/')[1];
+   const url = new URL(deepLinkUrl);
+   const postId = url.pathname.split('/')[2];
    if (DEEPLINK_POST_REGEX.test(deepLinkUrl) && postId) checkIsAuthorFollowingMe(postId);
  };

  const handleDeepLinkUser = async (deepLinkUrl) => {
-   const username = deepLinkUrl?.split('helio.social/')[1];
+   const url = new URL(deepLinkUrl);
+   const username = url.pathname.split('/')[1];
    if (username?.toLocaleLowerCase() === profile?.username?.toLocaleLowerCase()) {
      return navigation.navigate('Profile');
    }

    if (DEEPLINK_USER_REGEX.test(deepLinkUrl) && username) {
      try {
-       const replacedUsername = username?.replace('+', '');
+       const replacedUsername = decodeURIComponent(username);
        const data = {
          username: replacedUsername,
          user_id: signedProfileId
        };
        handleMovePage(USER, data);
      } catch (e) {
        console.error(e);
      }
    }
  };

  const handleDeepLinkCommunity = async (deepLinkUrl) => {
-   const communityName = deepLinkUrl?.split('helio.social/c/')[1];
+   const url = new URL(deepLinkUrl);
+   const communityName = url.pathname.split('/')[2];
    if (DEEPLINK_COMMUNITY_REGEX.test(deepLinkUrl) && communityName) {
      try {
        navigation.navigate('TopicPageScreen', {id: communityName});
      } catch (e) {
        console.error(e);
      }
    }
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->